### ISSUE TITLE
Is the grinder doing more work than required?

### ISSUE THREAD

**Issue by pshipton** [adoptium/aqa-tests#3986]
**Describe the bug**
I started a grinder using "NodesByIterations" on 5 machines. The parent job loads all the repos and downloads the JVM, only to create 5 child jobs that actually runs the testing. It seems loading the repos and downloading the JVM in the parent job is a waste of time.
https://openj9-jenkins.osuosl.org/job/Grinder/1292/console

**To Reproduce**
Steps to reproduce the behavior (or Grinder rerun link):

**Expected behavior**

**Screenshots**

**Additional context**

**Comment by llxia** [comment]
I agree. We only need to download JVM to run TKG when PARALLEL=Dynamic. In other parallel cases (i.e., NodesByIterations, Subdir), we do not need to download JVM and clone test repos.

**Comment by llxia** [comment]
As @AdamBrousseau pointed out, there is another corner case where PARALLEL=Dynamic and NUM_MACHINES > 1, but the available machine is only 1. In this case, we should run it serially (not launching the child job).
